### ISSUE TITLE
Add SNI/ProxyProtocol to Wazuh

### ISSUE THREAD

**Issue by francobep** [wazuh/wazuh#16199]
|Wazuh version|Component|Install type|
|---|---|---|
| 4.3.3 | Remoted | Manager |

The current agent reporting protocol does not support SNI and ProxyProtocol. Adding these capabilities allows a more versatile way to handle traffic between agents and managers.

**Comment by xBounceIT** [comment]
Im having issues trying to put traefik in front of multiple wazuh servers and make it understand when to send agent1 to wazuh1 and agent2 to wauzh2. Am i understanding right that the Wazuh agent doesn't send the SNI header?

**Comment by strowi** [comment]
Yes, please. Running on Kubernetes with Traefik-Ingress behind a Loadbalancer i only see the LBs IP in each request instead of the real client.

**Comment by xBounceIT** [comment]
up

**Comment by jeremy-cayrasso** [comment]
4.13.1 is still not sending the SNI, please add SNI support

**Comment by tobiasmeier78** [comment]
Is there a implementation roadmap? This feature would be very helpful.

**Comment by yggdrasilodin** [comment]
up

**Comment by mehrdad-khojastefar** [comment]
up
